### PR TITLE
turned (InverseOf)MorphismFromCoimageToImage into CAP operations

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Review by zickgraf — CHANGES_REQUESTED** [review]

File: CAP/gap/MethodRecord.gi [hunk 1]
```diff
@@ -2549,6 +2549,18 @@ MorphismFromCoimageToImageWithGivenObjects := rec(
...
+MorphismFromCoimageToImage := rec(
+  filter_list := [ "category", "morphism" ],
+  input_arguments_names := [ "cat", "alpha" ],
+  output_source_getter_string := "CoimageObject( cat, alpha )",
+  output_source_getter_preconditions := [ [ "CoimageObject", 1 ] ],
+  output_range_getter_string := "ImageObject( cat, alpha )",
+  output_range_getter_preconditions := [ [ "ImageObject", 1 ] ],
+  with_given_object_position := "both",
+  dual_operation := "MorphismFromCoimageToImage",
+  dual_arguments_reversed := true,
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi:2561]: What is superfluous? `dual_arguments_reversed := true`?

[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi:2561]: Done.

**Comment by mohamed-barakat** [comment]
Please do not merge until https://github.com/homalg-project/CategoricalTowers/pull/382 is merged.

**Comment by mohamed-barakat** [comment]
> Please do not merge until [homalg-project/CategoricalTowers#382](https://github.com/homalg-project/CategoricalTowers/pull/382) is merged.

Can be merged now if no further changes are required.

**Comment by mohamed-barakat** [comment]
Again, please do not merge until https://github.com/homalg-project/CategoricalTowers/pull/383 is merged.

**Review by zickgraf — DISMISSED** [review]

**Comment by zickgraf** [comment]
Thanks, you can merge when it suits CategoricalTowers.

**Comment by mohamed-barakat** [comment]
> Thanks, you can merge when it suits CategoricalTowers.

Thanks for the approval. I cannot merge without changing the policy since the requirement codecov/project failed. Could you merge, please?

**Comment by zickgraf** [comment]
Codecov is not an issue, it's the conversations not marked as resolved:

> 2 conversations must be resolved before merging.

However, I just noticed that you have put the non-WithGiven operations below the WithGiven. We usually do it the other way round. Can you swap them?

**Comment by mohamed-barakat** [comment]
> However, I just noticed that you have put the non-WithGiven operations below the WithGiven. We usually do it the other way round. Can you swap them?

Done.

File: CAP/gap/MethodRecord.gi [hunk 2]
```diff
@@ -2560,6 +2572,18 @@ InverseOfMorphismFromCoimageToImageWithGivenObjects := rec(
...
+InverseOfMorphismFromCoimageToImage := rec(
+  filter_list := [ "category", "morphism" ],
+  input_arguments_names := [ "cat", "alpha" ],
+  output_source_getter_string := "ImageObject( cat, alpha )",
+  output_source_getter_preconditions := [ [ "ImageObject", 1 ] ],
+  output_range_getter_string := "CoimageObject( cat, alpha )",
+  output_range_getter_preconditions := [ [ "CoimageObject", 1 ] ],
+  with_given_object_position := "both",
+  dual_operation := "InverseOfMorphismFromCoimageToImage",
+  dual_arguments_reversed := true,
```
[the remark this answers is not in the document]
Comment (zickgraf) [reply CAP/gap/MethodRecord.gi:2572]: Somehow `dual_arguments_reversed` has appeared again here and is missing now below.

[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi]: Yes, just spotted the asymmetry after force pushing. Awkward. Fixed now.

**Review by zickgraf — APPROVED** [review]

**Pull request merged**